ThirdParty/fmt/support/cmake/cxx14.cmake

```diff
@@ -2,9 +2,19 @@
 
 include(CheckCXXSourceCompiles)
 include(CheckCXXCompilerFlag)
+if (FMT_ASSUME_CPP14_SUPPORT)
+	set(CMAKE_REQUIRED_FLAGS )
+	 set(SUPPORTS_VARIADIC_TEMPLATES ON)
+	 set(SUPPORTS_INITIALIZER_LIST ON)
+	 set(SUPPORTS_ENUM_BASE ON)
+	 set(SUPPORTS_TYPE_TRAITS ON)
+	 set(SUPPORTS_USER_DEFINED_LITERALS ON)
+	return()
+endif()
+
 
 if (NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 14)
```

Comment:
Is this something that could be done right before the add_subdirectory for fmt, and if it sets the CMAKE_CXX_STANDARD then unset it afterward?

I don't remember if we had issues with it, but something like in the fmt cxx14.cmake file:
```
if (NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
```
might be something to put in the HELICS CMakeLists.txt

Reply:
the CMAKE_CXX_STANDARD flag doesn't work correctly with all combinations of compilers and cmake versions so I haven't found it reliable.

Reply:
There were a few other things in the cmake for fmt that could be useful particularly for addressing #499.